config/config.go

```diff
@@ -238,6 +244,30 @@ func (c Config) FindComponent(name string) *resource.Config {
 	return nil
 }
 
+// SetUnprocessedConfig sets unprocessedConfig with a copy of the config passed in.
+func (c *Config) SetUnprocessedConfig(cfg *Config) error {
+	cpy, err := cfg.CopyOnlyPublicFields()
```

Comment:
I'm assuming `json.MarshalIndent` only writes public fields and therefore we're confident `CopyOnlyPublicFields` is the only thing stuff we need to pull out into a copy?

Having `SetUnprocessedConfig` doing the marshalling would be obviously equivalent to me. But happy to just get an OK and not worry about it.